### ISSUE TITLE
Worse image quality with tiles cut using rio-tiler than GDAL

### ISSUE THREAD

**Issue by Plantain** [cogeotiff/rio-tiler#662]
When I use rio-tiler directly on (space-projected) tiffs, the quality of the tiles is noticeably worse than if I first cut them to GoogleMapsCompatible COG's using gdal.

I think given GDAL is the backend of rasterio/rio-tiler, and it's cutting the same size output tiles, the result should be the same quality if not pixel-identical.

Data: http://static.skysight.io/rio-bug/orig.tiff
Warp command: gdalwarp -multi -co TILED=YES -wo NUM_THREADS=64 -co BIGTIFF=YES -t_srs EPSG:3857 -of COG -co QUALITY=100 -co COMPRESS=JPEG -co TILING_SCHEME=GoogleMapsCompatible -co ZOOM_LEVEL=7 -co BLOCKSIZE=512 --config GDAL_TIFF_OVR_BLOCKSIZE 512 orig.tiff reproj.tiff

Code:
```
from rio_tiler.io import Reader
with Reader("orig.tiff") as src:
    img = src.tile(226,162,8,512)
    f = open("orig.png", "wb")
    f.write(img.render())

with Reader("reproj.tiff") as src:
    img = src.tile(226,162,8,512)
    f = open("reproj.png", "wb")
    f.write(img.render())
```

rio-tiler reprojected and extracted tile:
![orig](https://github.com/cogeotiff/rio-tiler/assets/319895/dfd174f9-e3f8-474a-9fe5-3f021021f9fb)

gdal reprojected and rio-tiler extracted tile:
![reproj](https://github.com/cogeotiff/rio-tiler/assets/319895/50a31d83-5d79-4f89-86c9-e41f89764138)

The difference is subtle but if you zoom right in, you can see the GDAL reprojected tile appears to have double(?) the vertical pixel resolution.

![Screenshot 2023-12-14 at 10 59 40 am](https://github.com/cogeotiff/rio-tiler/assets/319895/c1b6cee9-25ec-44cc-8fb8-1c970f492e2d)
![Screenshot 2023-12-14 at 10 59 46 am](https://github.com/cogeotiff/rio-tiler/assets/319895/5c9b480c-4d7d-4ab3-a14a-48a7c6638d44)

rio-tiler 6.2.3-post1
rasterio 1.3.9 but built with GDAL 3.8.1
GDAL 3.8.1 for the projection
OSX 14.1.2

**Comment by vincentsarago** [comment]
@Plantain I wouldn't compare GDAL COG with rio-tiler but you can compare the output of gdalwarp command to read part of your file 

```python
import os
from rio_tiler.io import Reader


with Reader("http://static.skysight.io/rio-bug/orig.tiff") as src:
    img = src.tile(226, 162 ,8, tilesize=512, reproject_method="nearest", resampling_method="nearest")
    with open("226-162-8_RIOTILER.png", "wb") as f:
        f.write(img.render())

os.system("gdalwarp -of PNG -q /vsicurl/http://static.skysight.io/rio-bug/orig.tiff 226-162-8_GDAL.png -t_srs EPSG:3857 -ts 512 512 -te 15341217.324948 -5479006.187481 15497760.358876 -5322463.153553 -overwrite")
```

#### GDAL
![226-162-8_GDAL](https://github.com/cogeotiff/rio-tiler/assets/10407788/7f8f0b46-0240-4bc8-af64-3e272c00d90c)

#### rio-tiler
![226-162-8_RIOTILER](https://github.com/cogeotiff/rio-tiler/assets/10407788/c38d61d5-9c4b-4136-8c66-9a674399de75)



ahhhhhh this is because of #654 if you use 6.2.2 we get similar output 

![226-162-8_GDAL](https://github.com/cogeotiff/rio-tiler/assets/10407788/25a09ea4-940c-4647-8c9e-901c4315048f)
![226-162-8_RIOTILER](https://github.com/cogeotiff/rio-tiler/assets/10407788/4d6f2fcb-a651-4bd8-822a-671eca833985)

**Comment by vincentsarago** [comment]
@Plantain I'm going to closes because it's related to #654

**Comment by vincentsarago** [comment]
#654 was only resolved in 6.2.7 so any version from 6.2.3 to 6.2.7 would have the same behaviour
